### PR TITLE
Clean up data format of SSB population dataset

### LINKED ISSUE #94: SSB Population Data is hard to understand/work with
### Description
Right now, the SSB population data _data source_ is quite hard to understand and work with, and thus I believe (emphasis: _I_. Thus the `question` label – please share your thoughts) it might be helpful to do some cleanup of the data before sending it over to frontend.

If I'm not mistaken, the data is on GeoJSON format, which means we might be able to abstract this cleanup and reuse it later.

#### Proposed actions
- Provide a refined query which allows you to request the population for several municipalities using e.g. an array of enum values; `[Municipality.Oslo, Municipality.Trondheim]` or something along those lines. This requires the creation of an enum with all the municipalities, but I believe this should be doable with some minimal scripting in python. Rumor has it @Kpaubert is a God at this
- Clean up the data so that the returned data to frontend is similar to `{ municipalities: [...], populations: [...] }`. This would make it a lot easier to chart

### Subtasks
- [x] Decide what to do. Paging @IT2901-SINTEF01/frontend for discussion; how would you like the data to look? 
- [x] Implement eventual changes
- [x] Test!

### PR DESCRIPTION
**Pull request by FredrikAugust**
Closes #94 

```graphql
query {
	populationsInNorway {
		# municipalitiesWithKeys
		dataset(years: ["2018", "2019", "2020", "2021"], municipalities: ["K-0301", "K-3024"]) {
			value {
				municipality
				populationForYear {
					year
					population
				}
			}
		}
	}
}
```

You can now use this format when requesting data, which is a bit easier to read and work with for @IT2901-SINTEF01/frontend.

### CODE REVIEW

File: API/Queries/RootQuery.cs [hunk 1]
```diff
@@ -1,8 +1,9 @@
 ﻿using System;
+using System.Collections.Generic;
```
Comment (Kpaubert) [inline API/Queries/RootQuery.cs:2]: This import is not used.

File: API/Queries/RootQuery.cs [hunk 2]
```diff
@@ -14,7 +15,8 @@ public RootQuery(IServiceProvider serviceProvider)
         {
             var metAPIService = (IMetAPIService) serviceProvider.GetService(typeof(IMetAPIService));
             var metadataService = (IMetadataService) serviceProvider.GetService(typeof(IMetadataService));
-            var populationService = (IPopulationInNorwayService) serviceProvider.GetService(typeof(IPopulationInNorwayService));
+            var populationService =
+                (IPopulationInNorwayService) serviceProvider.GetService(typeof(IPopulationInNorwayService));
 
             Field<ForecastType>("forecast", "Latitude and Longitude defaults to the coordinates of Trondheim",
```
Comment (Kpaubert) [inline API/Queries/RootQuery.cs:21]: Not related to the PR, so this is just a note: this description does not describe the functionality of the query.

File: BackendTests/UnitTests/PopulationInNorwayDatasetTypeTests.cs [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    public static class PopulationInNorwayDatasetTypeTests
+    {
+        [Fact]
+        public static void PopulationInNorwayDatasetCorrectOutput()
+        {
+            var actor = new PopulationInNorwayDatasetType();
+
+            Assert.NotNull(actor.GetField("value"));
+
+            var expected = new LabeledValue()
```
Comment (Kpaubert) [inline BackendTests/UnitTests/PopulationInNorwayDatasetTypeTests.cs:23]: When not using `new()` _and_ using curly brackets to set the properties, the parenthesis are redundant when instantiating an object and can be removed. Change or let stay as is, I don't have any preference.

File: BackendTests/UnitTests/PopulationInNorwayTypeTests.cs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+using Backend.Models.SSBPopulationStatistics.GraphQLTypes;
+using GraphQL;
+using Shouldly;
+using Xunit;
+
+namespace BackendTests.UnitTests
+{
+    public sealed class PopulationInNorwayTypeTests
+    {
+        private readonly PopulationInNorwayType objectGraph = new(new MetadataServiceMocked());
```
Comment (Kpaubert) [inline BackendTests/UnitTests/PopulationInNorwayTypeTests.cs:13]: Let's get an underscore on this one (love the instantiation though `new(new ...)`).

File: BackendTests/UnitTests/PopulationInNorwayTypeTests.cs [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+            
+            var result = (IEnumerable<List<string>>) field.Resolver.Resolve(new ResolveFieldContext());
+            var list = result.ToList();
+            
+            list[0][0].ShouldBe("K-3001");
+            list[0][1].ShouldBe("Halden");
+        }
+
+        [Fact]
+        public void Years()
```
Comment (Kpaubert) [inline BackendTests/UnitTests/PopulationInNorwayTypeTests.cs:29]: This is not a clear name.

File: BackendTests/UnitTests/PopulationInNorwayTypeTests.cs [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+using Xunit;
+
+namespace BackendTests.UnitTests
+{
+    public sealed class PopulationInNorwayTypeTests
+    {
+        private readonly PopulationInNorwayType objectGraph = new(new MetadataServiceMocked());
+        
+        [Fact]
+        public void MunicipalitiesWithKeys()
```
Comment (Kpaubert) [inline BackendTests/UnitTests/PopulationInNorwayTypeTests.cs:16]: Clarify what the test tests.

File: Models/Base/JsonStat/JsonStatDatasetType.cs [hunk 7]
```diff
@@ -2,14 +2,22 @@
 
 namespace Backend.Models.Base.JsonStat
 {
-    public class JsonStatDatasetType<T, TU> : ObjectGraphType<JsonStatDataset<T>> where T : AbstractJsonStatDimension
+    public abstract class JsonStatDatasetType<T, TU> : ObjectGraphType<JsonStatDataset<T>>
```
Comment (Kpaubert) [inline Models/Base/JsonStat/JsonStatDatasetType.cs:5]: With several generic types, I think it's preferable to name them `<T1, T2>`, or `<T, U>`.

File: Models/MetAPI/GraphQLTypes/ForecastType.cs [hunk 8]
```diff
@@ -1,8 +1,6 @@
 ﻿using Backend.API.Services;
-using Backend.Models.Base.Metadata.GraphQLTypes;
 using Backend.Models.MetAPI.POCO;
 using Backend.utils.GraphQLTypes;
-using GraphQL.Types;
 
 namespace Backend.Models.MetAPI.GraphQLTypes
```
Comment (Kpaubert) [inline Models/MetAPI/GraphQLTypes/ForecastType.cs:5]: Not changed in your PR, but perhaps assign the nullable field and type field for the last field in the ForecastType?

File: Models/SSBPopulationStatistics/GraphQLTypes/PopulationForYearType.cs [hunk 9]
```diff
@@ -0,0 +1,14 @@
+using Backend.Models.SSBPopulationStatistics.POCO;
+using GraphQL.Types;
+
+namespace Backend.Models.SSBPopulationStatistics.GraphQLTypes
+{
+    public class PopulationForYearType : ObjectGraphType<PopulationForYear>
```
Comment (Kpaubert) [inline Models/SSBPopulationStatistics/GraphQLTypes/PopulationForYearType.cs:6]: Unclear name imo, is it population for a given year? Population per year?

File: Models/SSBPopulationStatistics/GraphQLTypes/LabeledValueType.cs [hunk 10]
```diff
@@ -0,0 +1,14 @@
+using Backend.Models.SSBPopulationStatistics.POCO;
+using GraphQL.Types;
+
+namespace Backend.Models.SSBPopulationStatistics.GraphQLTypes
+{
+    public class LabeledValueType : ObjectGraphType<LabeledValue>
```
Comment (Kpaubert) [inline Models/SSBPopulationStatistics/GraphQLTypes/LabeledValueType.cs:6]: This naming seems sort of unclear for the untrained eye, perhaps add msdocs on this?

File: Models/SSBPopulationStatistics/GraphQLTypes/PopulationInNorwayDatasetType.cs [hunk 11]
```diff
@@ -0,0 +1,45 @@
+using System;
```
Comment (Kpaubert) [inline Models/SSBPopulationStatistics/GraphQLTypes/PopulationInNorwayDatasetType.cs:1]: Not used.

File: Models/SSBPopulationStatistics/GraphQLTypes/PopulationInNorwayDatasetType.cs [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+            Field<ListGraphType<LabeledValueType>>("value", "The values for the selected municipalities and years.",
+                null,
+                context =>
+                {
+                    var municipalities = context.Parent.GetArgument<List<string>>("municipalities");
+                    var years = context.Parent.GetArgument<List<string>>("years");
+
+                    var municipalityEntrySize = context.Source.Dimension.Size[1];
+
+                    return (from municipality in municipalities
```
Comment (Kpaubert) [inline Models/SSBPopulationStatistics/GraphQLTypes/PopulationInNorwayDatasetType.cs:28]: Nice :awooga:

**Review by Kpaubert — CHANGES_REQUESTED** [review]
Couple o'comments, looks good overall.

File: BackendTests/UnitTests/PopulationInNorwayTypeTests.cs (hunk 5 above)
Comment (FredrikAugust) [reply BackendTests/UnitTests/PopulationInNorwayTypeTests.cs:29]: Yes!

**Review by Kpaubert — APPROVED** [review]
Nice!

**Comment by FredrikAugust** [comment]
Admin force rebase as the review dismissal only happened due to comment changes.

**Pull request merged**